### ISSUE TITLE
BOSS fails when zerg upgrades their main hatchery to lair, then wants to build an expansion.

### ISSUE THREAD

**Issue by earthiverse** [davechurchill/ualbertabot#19]
Look, it upgraded!
![scscrnshot_112115_203623](https://cloud.githubusercontent.com/assets/602905/11321935/d37d6e8e-908f-11e5-9b9a-46f9e0398603.gif)

Then if I manually take a drone, and make a hatchery, like my build order goal wants, things go back to normal.
![scscrnshot_112115_203630](https://cloud.githubusercontent.com/assets/602905/11321937/dfe51e6a-908f-11e5-9d8f-5fc2873cfa33.gif)

It also doesn't work if I make the goal 1 lair and 1 hatchery.
![scscrnshot_112115_204505](https://cloud.githubusercontent.com/assets/602905/11321948/d2d03a56-9090-11e5-91b7-6d394fb0dd64.gif)

It also doesn't work if I make the goal 2 hatcheries.
![2-hatcheries](https://cloud.githubusercontent.com/assets/602905/11321945/5c541316-9090-11e5-8b24-5089b221e384.gif)

**Comment by davechurchill** [comment]
I pushed what I thought was a fix to this a few days ago, did you try it out?

**Comment by earthiverse** [comment]
Yes, I did [pull that commit](https://github.com/earthiverse/ualbertabot-micro_buro/commit/0b36daf336f3daaf6576de61417c15c16bd441aa), but it seems like it still has an issue.

Before it would error and say "No Depots :("; now it says "NBOS Solution".

**Comment by davechurchill** [comment]
Ok, will look into it but can't promise anything in the next few days

**Comment by earthiverse** [comment]
I don't know if it's related, but I keep an extra hatchery around at all times to deal with this bug in the meantime.

I put in num_hatcheries + 1 as my goal, with the same number of lairs as well, but it will never go to build a hatchery, ever.

If we use the zergling build, and it ends up expanding, it expands no problem, as it only builds hatcheries.

![scscrnshot_120115_181008](https://cloud.githubusercontent.com/assets/602905/11519357/ce034d38-9856-11e5-9e61-ee0f3eda6403.gif)
The only two resource depots are in the screenshot, and there's nothing in the build order queue. But you can see the 2 hatcheries and 1 lair in the build order search.

**Comment by davechurchill** [comment]
Sorry I wasn't able to fix it recently, been very busy. Good that you found a workaround for now
